### PR TITLE
Consistent line height, other style fixes, better style guide

### LINKED ISSUE #249: Better distinguish <dt> from <dd>
Adam:

> img
> 
> I don't really have any great ideas here, but this is pretty weird-looking, especially when a user's answers contain line breaks.

### PR DESCRIPTION
**Pull request by jrubenoff**
Closes #249, #241, #223, #204, and dobtco/screendoor-v2#3302

- Uses a global unitless `line-height`, and makes appropriate alignment adjustments

----

Also:

- `<dt>` is now bold for added clarity:

screen shot 2016-04-01 at 1 30 24 pm

- Improved "dos and don'ts" pattern, and nicer-looking screenshots, in the style guide:

screen shot 2016-04-01 at 1 31 03 pm

- Tighter spacing for formatted lists:

screen shot 2016-04-01 at 1 32 16 pm

- New documentation for text contrast, small caps, links to external sites in dropdowns

### CODE REVIEW

**Comment by ajb** [comment]
Can you talk more about the decision to add units to the global line height? I'm trying to square it with https://css-tricks.com/almanac/properties/l/line-height/#article-header-id-0.

Maybe we should discuss on Monday.

**Comment by jrubenoff** [comment]
We use a set, predictable range of font sizes that are hardcoded into variables, so that rule doesn't really apply. If we had a bunch of exceptions in our codebase where we were styling elements in arbitrary font sizes and prioritized flexibility over predictability, that advice would make sense. The approach in this PR gives us greater control over the design.

We were already using set line heights for `h1`, `h2`, `h3`, etc., so this just extends that pattern to the whole codebase.

**Comment by jrubenoff** [comment]
On call:

- [ ] Remove units from all line heights

**Comment by jrubenoff** [comment]
Alright, removed the units from all the line-heights. `1.5` is a little too tight for our smaller sizes, so I adjusted the `$lineHeightSmall-` and `$unitlessLineHeightSmall-` variables accordingly. Also made a few more minor improvements to the style guide.

File: vendor/assets/stylesheets/dvl/components/page_header.scss [hunk 1]
```diff
@@ -99,13 +99,12 @@
 .page_header_context {
   width: 100%;
   position: relative;
-  top: $rhythm * 0.5;
+  top: $lineHeightSmall / 2;
   left: $fontOpticalMarginH2 * -1;
```
Comment (ajb) [inline vendor/assets/stylesheets/dvl/components/page_header.scss]: I think we can change this to `left: 0`, based on what it looks like.

File: vendor/assets/stylesheets/dvl/components/navbar.scss [hunk 2]
```diff
@@ -110,9 +110,8 @@
   text-align: center;
   display: inline-block;
   padding: 0 $rhythm / 2;
-  height: $rhythm * 2;
-  line-height: $rhythm * 2;
-  font-size: $fontSmallest;
+  height: $lineHeightSmallest;
+  @include fontSmallestBody;
```
Comment (ajb) [inline vendor/assets/stylesheets/dvl/components/navbar.scss]: Here's how this badge looks in `master`:

screen shot 2016-05-05 at 7 42 19 am

Here's how it looks here:

screen shot 2016-05-05 at 7 42 40 am

:'(

File: vendor/assets/stylesheets/dvl/components/sidebar_data.scss [hunk 3]
```diff
@@ -44,15 +45,22 @@
 }
 
 .sidebar_data_icon {
-  float: left;
-  width: 1rem;
+  position: absolute;
+  top: 50%;
+  left: 0;
+  margin-top: $rhythm * -3;
```
Comment (ajb) [inline vendor/assets/stylesheets/dvl/components/sidebar_data.scss]: Don't we have a mixin for this?

Comment (ajb) [inline vendor/assets/stylesheets/dvl/components/sidebar_data.scss]: The font size is `2.5rem`, so I believe the `margin-top` needs to be `-1.25rem`, not `-1.5rem` as it is.

File: vendor/assets/stylesheets/dvl/components/sidebar_data.scss [hunk 4]
```diff
@@ -44,15 +45,22 @@
...
   i {
     line-height: 1;
-    color: $darkerGray;
+    color: $gray;
   }
   @media screen and (min-width: $lapWidth) {
     width: $rhythm * 5;
+    left: $rhythm * 2;
+    .sidebar_data li:last-of-type & {
+      margin-top: $rhythm * -1.5;
```
Comment (ajb) [inline vendor/assets/stylesheets/dvl/components/sidebar_data.scss]: I can't figure this out by looking at it. Explain?

File: vendor/assets/stylesheets/dvl/core/forms.scss [hunk 5]
```diff
@@ -463,17 +463,19 @@ form {
 }
 
 .form_error {
-  font-size: $fontSmall;
+  @include fontSmallBody;
   background-color: $errorColor;
   color: $white;
-  padding: $rhythm $rhythm * 2;
+  padding: ($rhythm / 2) ($rhythm * 1.5);
```
Comment (ajb) [inline vendor/assets/stylesheets/dvl/core/forms.scss]: Much better :D

**Comment by ajb** [comment]
Here are my comments as I went through the changed files and previewed the site. I also added some line comments where appropriate.

I think a good next step would be to talk in-person (or video) before making any more changes.

---

Contrary to https://github.com/dobtco/screendoor-v2/pull/3523#issuecomment-216679696, adding lines of code _does_ seem to be a trend: http://take.ms/hAGhF. This is my primary resistance to this PR.

Similarly, I'm not sure the 1.64 line height is worth the added complexity, the likelihood that it won't be applied consistently, and the added file size.

And while we're on the subject: these large, multi-issue PRs are really a nightmare to deal with. We've talked about this before; how can we avoid them in the future?

---

I found what I think is a page header alignment bug: http://take.ms/ZZYVS

I might have missed it... why are "dos" and "don'ts" screenshots, instead of live HTML? The screenshots look kinda crappy on my display... I think because they're non-retina?

This footer line-height bug still exists: http://take.ms/ctfCv

The sidebar has some behavior that we've tried to avoid in the past: http://take.ms/ktK3X. (This might be why: http://take.ms/USHFY)

The redesigned user block looks nice, but feels a little cramped. How about adding some more padding on the right side? http://take.ms/lripS

---

**Comment by jrubenoff** [comment]
I don't share your concerns around the `page_header` and `sidebar_data`, but I made a few aesthetic changes to those components. I also reverted the line-height mixins.

**Comment by ajb** [comment]
@jrubenoff: there are some merge conflicts that need to be fixed, now that I merged the other footer line height PR and the list styles PR.

Can you do that, and then maybe write up a new summary for this PR? I'm having trouble actually fguring out what this changes.

**Comment by jrubenoff** [comment]
Done and done

File: vendor/assets/stylesheets/dvl/components/sidebar_data.scss (hunk 3 above)
Comment (ajb) [inline vendor/assets/stylesheets/dvl/components/sidebar_data.scss]: @jrubenoff: did you see this?

File: vendor/assets/stylesheets/dvl/components/sidebar_data.scss (hunk 4 above)
Comment (ajb) [inline vendor/assets/stylesheets/dvl/components/sidebar_data.scss]: @jrubenoff: would appreciate if you can explain

File: vendor/assets/stylesheets/dvl/components/tooltips.scss [hunk 6]
```diff
@@ -10,7 +10,6 @@
   min-width: 40px;
   transition: transform 0.25s $easeOutBack, opacity 120ms ease-out;
   // Override styles on parent links
-  line-height: $lineHeight;
```
Comment (ajb) [inline vendor/assets/stylesheets/dvl/components/tooltips.scss]: I'm pretty sure this was important to have in here, as per the comment on line 12.

**Comment by ajb** [comment]
Here's my list of what components changed:

- dropdown select
- dropdowns
- flashes
- navbar badge
- page header
- page header context
- page header tertiary
- sidebar data
- sidebar nav
- sidebar sub actions
- tooltips
- user block
- datetime picker
- `<dt>` smaller


I'm gonna preview all of these and look at the changes. I also left some new inline comments.

**Comment by ajb** [comment]
Here's all I have for now:

- img
- [ ] Should the `<dt>` really be `$fontSmallest`? I thought we should use it sparingly.

Thanks for sticking with this... 😅

File: vendor/assets/stylesheets/dvl/components/sidebar_data.scss (hunk 3 above)
Comment (jrubenoff) [inline vendor/assets/stylesheets/dvl/components/sidebar_data.scss]: No, thanks for bringing it to my attention.

File: vendor/assets/stylesheets/dvl/components/tooltips.scss (hunk 6 above)
Comment (jrubenoff) [inline vendor/assets/stylesheets/dvl/components/tooltips.scss]: Thanks, fixed.

**Comment by jrubenoff** [comment]
Manually set the line-height for `navbar_badge`

`<dt>`s are `$fontSmaller`... maybe you misread?

File: vendor/assets/stylesheets/dvl/components/sidebar_data.scss (hunk 4 above)
Comment (jrubenoff) [inline vendor/assets/stylesheets/dvl/components/sidebar_data.scss]: The last sidebar item has no bottom padding, and the first sidebar item has no top padding. So, we need to adjust the position of the icon to vertically align it properly for each item.

Comment (jrubenoff) [inline vendor/assets/stylesheets/dvl/components/sidebar_data.scss]: I'll add a comment explaining this.

**Comment by ajb** [comment]
> <dt>s are $fontSmaller... maybe you misread?

Yeah, I did. But did they get smaller in this PR?

**Comment by jrubenoff** [comment]
Yes, they were `fontSmall` before. Harder to distinguish.

**Pull request merged**